include/util/graph_loader.hpp

```diff
@@ -50,26 +49,26 @@ NodeID loadNodesFromFile(storage::io::FileReader &file_reader,
     coordinates.resize(number_of_nodes);
     osm_node_ids.reserve(number_of_nodes);
 
-    extractor::ExternalMemoryNode current_node;
+    extractor::QueryNode current_node;
     for (NodeID i = 0; i < number_of_nodes; ++i)
     {
         file_reader.ReadInto(&current_node, 1);
 
         coordinates[i].lon = current_node.lon;
         coordinates[i].lat = current_node.lat;
         osm_node_ids.push_back(current_node.node_id);
+    }
 
-        if (current_node.barrier)
-        {
-            *barriers = i;
-            ++barriers;
-        }
-
-        if (current_node.traffic_lights)
-        {
-            *traffic_signals = i;
-            ++traffic_signals;
-        }
+    auto num_barriers = file_reader.ReadElementCount64();
+    for (auto index = 0UL; index < num_barriers; ++index)
+    {
```

Comment:
As above - might be a bit more performant to read larger blocks.  That said, there probably aren't very many of these in the grand scheme of things, so maybe not worth the extra logic.